api-server/api-server-common/src/storage/impls/in_memory/mod.rs

```diff
@@ -801,6 +814,19 @@ impl ApiServerInMemoryStorage {
         Ok(())
     }
 
+    fn set_order_at_height(
+        &mut self,
+        order_id: OrderId,
+        order: &Order,
```

Comment:
For a setter, it's more natural to pass the value "by value" and clone outside if needed.